src/main/java/org/gaul/s3proxy/EncryptedBlobStore.java

```diff
@@ -0,0 +1,784 @@
+                long size = blobMeta.getSize() - partPaddingSizes;
+                mbm.setSize(size);
+                mbm.getContentMetadata().setContentLength(size);
+            } else {
+                long size = blobMeta.getSize() - Constants.PADDING_BLOCK_SIZE;
+                mbm.setSize(size);
+                mbm.getContentMetadata().setContentLength(size);
+            }
+        }
+
+        return mbm;
+    }
+
+    private boolean multipartRequiresStub() {
+        String blobStoreType = getBlobStoreType();
+        return Quirks.MULTIPART_REQUIRES_STUB.contains(blobStoreType);
+    }
+
+    private String blobNameWithSuffix(String container, String name) {
+        String nameWithSuffix = blobNameWithSuffix(name);
+        if (delegate().blobExists(container, nameWithSuffix)) {
+            name = nameWithSuffix;
+        }
+        return name;
+    }
+
+    private String blobNameWithSuffix(String name) {
+        return name + Constants.S3_ENC_SUFFIX;
+    }
+
+    private String getBlobStoreType() {
+        return delegate().getContext().unwrap().getProviderMetadata().getId();
+    }
+
+    private String generateUploadId(String container, String blobName) {
+        String path = container + "/" + blobName;
+        return DigestUtils.sha256Hex(path);
+    }
+
+    @Override
+    public Blob getBlob(String containerName, String blobName) {
+        try {
+            // adjust name if a blob with the .s3enc suffix exists
+            blobName = blobNameWithSuffix(containerName, blobName);
+
+            // init the decryption for a full blob
+            Decryption decryption =
+                new Decryption(secretKey, delegate(), containerName, blobName,
+                    0L, -1L);
+            return decryptBlob(decryption, containerName,
+                delegate().getBlob(containerName, blobName));
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public Blob getBlob(String containerName, String blobName,
+        GetOptions getOptions) {
+
+        try {
+            // adjust name if a blob with the .s3enc suffix exists
+            blobName = blobNameWithSuffix(containerName, blobName);
+
+            // init the decryption
+            Decryption decryption = null;
+            if (getOptions != null && getOptions.getRanges().size() > 0) {
+
+                // we support only one range
+                if (getOptions.getRanges().size() > 1) {
+                    Exception e =
+                        new Exception("multiple ranges are not supported");
+                    throw new RuntimeException(e);
+                }
+
+                String range = getOptions.getRanges().get(0);
+                String[] ranges = range.split("-", 2);
+                if (ranges[0].isEmpty()) {
+                    // clear current ranges to avoid multiple ranges
+                    getOptions.getRanges().clear();
```

Comment:
Hoist this above the `if` statement -- we have to update the range header regardless.